### DISCUSSION TITLE
Allow relative dates on timedate fields

### DISCUSSION THREAD

**Discussion by beatbrot** [ankitpokhrel/jira-cli#708] [Ideas]
I want to be able to specify datetimes via relative strings (like "-1d", "-1h", etc...)

So for example, if I want to add a worklog to an issue with a worklog starte datetime, I want to be able to run:

```sh
jira issue worklog add ABC-123 --started -1d
```

**Comment by ankitpokhrel** [comment]
Hi @beatbrot, the Jira API expects date time for that field unlike some other fields where a relative time would work. I am not inclined towards adding this transformation logic to the tool unless absolutely necessary. Perhaps we can manage with the standard date-time format for this?

![Screen Shot 2024-02-06 at 7 17 03 PM](https://github.com/ankitpokhrel/jira-cli/assets/2364546/ad670cd9-a8d1-4ed8-bf35-7ea77a92424f)

**Reply by beatbrot** [reply]
Yeah, I understand that that's not what the JIRA API expects. My intention would be exactly that - add the transformation logic to the tool.

But I understand your concern, and I am not sure whether I would add this feature as a developer. But as a user, I'd love to have it 😉. But in the end, it's your decision 😄
